operator/pkg/model/translation/redirect_test.go

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package translation
+
+import (
+	"testing"
+
+	envoy_config_route_v3 "github.com/cilium/proxy/go/envoy/config/route/v3"
+	envoy_type_matcher_v3 "github.com/cilium/proxy/go/envoy/type/matcher/v3"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cilium/cilium/operator/pkg/model"
+)
+
+func TestRequestRedirectSchemeHeaderMatcher(t *testing.T) {
```

Comment:
This test is not actually testing the correct code; we should be testing `envoyHTTPRoutes()`. This tests that the logic included in the changes produces some output, not that the code actually works how it should.

This code will continue to pass even if breaking changes are made to `envoyHTTPRoutes()`.

Please update this to test `envoyHTTPRoutes()` instead. Check `envoy_virtual_host_test.go` for similar patterns.